src/sql/src/plan/typeconv.rs

```diff
@@ -337,13 +337,41 @@ fn get_cast(
 ) -> Option<Cast> {
     use CastContext::*;
 
+    // Determines if types are equal indepdendent of any custom types.
```

Comment:
nit: what do you think about calling this "structural equality"? I suspect this will come up a bunch, and "structural equality" is the term of art for this.

Reply:
Oh perfect.